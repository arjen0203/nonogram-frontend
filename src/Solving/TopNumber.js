import React from "react";


class TopNumber extends React.Component {
    render() {
        var styling;
        if (this.props.value.holds === false){
            styling = "to-be-solved-top";
        } else {
            styling = "solved-top"
        }

        return (
            <div className={styling}>{this.props.value.number}</div>
        );
    }
}

export default TopNumber;