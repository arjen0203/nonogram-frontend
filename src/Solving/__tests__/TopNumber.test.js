import React from "react";
import ReactDOM from 'react-dom';
import TopNumber from "../TopNumber";
import {render, cleanup} from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect"

import renderer from "react-test-renderer";

afterEach(cleanup);

it('Renders without crashing', function () {
	const val = {number: 6, holds: false};

	const div = document.createElement('div');
	ReactDOM.render(<TopNumber value={val}></TopNumber>, div);
});

it('Render correctly', function () {
	const val = {number: 6, holds: false};

	const comp = render(<TopNumber value={val}></TopNumber>).getByTestId("topNumber");
	expect(comp).toHaveTextContent("6");
	expect(comp).toHaveClass("to-be-solved-top");
});

it('Render correctly 2', function () {
	const val = {number: 3, holds: false};

	const comp = render(<TopNumber value={val}></TopNumber>).getByTestId("topNumber");
	expect(comp).toHaveTextContent("3");
	expect(comp).toHaveClass("to-be-solved-top");
});

it('Render correctly is holding', function () {
	const val = {number: 6, holds: true};

	const comp = render(<TopNumber value={val}></TopNumber>).getByTestId("topNumber");
	expect(comp).toHaveTextContent("6");
	expect(comp).toHaveClass("solved-top");
});

it('Render correctly is holding 2', function () {
	const val = {number: 3, holds: true};

	const comp = render(<TopNumber value={val}></TopNumber>).getByTestId("topNumber");
	expect(comp).toHaveTextContent("3");
	expect(comp).toHaveClass("solved-top");
});

it('matches snapshot', function () {
	const val = {number: 6, holds: false};

	const tree = renderer.create(<TopNumber value={val}></TopNumber>).toJSON();
	expect(tree).toMatchSnapshot();
});