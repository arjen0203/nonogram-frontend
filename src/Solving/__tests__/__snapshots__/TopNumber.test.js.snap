// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="to-be-solved-top"
  data-testid="topNumber"
>
  6
</div>
`;
